### PR TITLE
MAPREDUCE-7411: use secure XML parsers

### PR DESCRIPTION
**Pull request by pjfanning**
### For code changes:

- [x] Does the title or this PR starts with the corresponding JIRA issue id (e.g. 'HADOOP-17799. Your PR title ...')?
- [ ] Object storage: have the integration tests been executed and the endpoint declared according to the connector-specific documentation?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the `LICENSE`, `LICENSE-binary`, `NOTICE-binary` files?

### CODE REVIEW

**Comment by hadoop-yetus** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 54s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  1s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  1s |  |  detect-secrets was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  |  The patch appears to include 12 new or modified test files.  |
|||| _ trunk Compile Tests _ |
| +0 :ok: |  mvndep  |  16m  3s |  |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |  28m 53s |  |  trunk passed  |
| +1 :green_heart: |  compile  |   2m 48s |  |  trunk passed with JDK Ubuntu-11.0.16+8-post-Ubuntu-0ubuntu120.04  |
| +1 :green_heart: |  compile  |   2m 29s |  |  trunk passed with JDK Private Build-1.8.0_342-8u342-b07-0ubuntu1~20.04-b07  |
| +1 :green_heart: |  checkstyle  |   1m 19s |  |  trunk passed  |
| +1 :green_heart: |  mvnsite  |   3m 25s |  |  trunk passed  |
| +1 :green_heart: |  javadoc  |   2m 46s |  |  trunk passed with JDK Ubuntu-11.0.16+8-post-Ubuntu-0ubuntu120.04  |
| +1 :green_heart: |  javadoc  |   2m 43s |  |  trunk passed with JDK Private Build-1.8.0_342-8u342-b07-0ubuntu1~20.04-b07  |
| +1 :green_heart: |  spotbugs  |   5m 24s |  |  trunk passed  |
| +1 :green_heart: |  shadedclient  |  21m 40s |  |  branch has no errors when building and testing our client artifacts.  |
|||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 31s |  |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   2m 17s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   2m 31s |  |  the patch passed with JDK Ubuntu-11.0.16+8-post-Ubuntu-0ubuntu120.04  |
| +1 :green_heart: |  javac  |   2m 31s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   2m 14s |  |  the patch passed with JDK Private Build-1.8.0_342-8u342-b07-0ubuntu1~20.04-b07  |
| +1 :green_heart: |  javac  |   2m 14s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| +1 :green_heart: |  checkstyle  |   0m 56s |  |  hadoop-mapreduce-project/hadoop-mapreduce-client: The patch generated 0 new + 32 unchanged - 1 fixed = 32 total (was 33)  |
| +1 :green_heart: |  mvnsite  |   2m 27s |  |  the patch passed  |
| +1 :green_heart: |  javadoc  |   1m 45s |  |  the patch passed with JDK Ubuntu-11.0.16+8-post-Ubuntu-0ubuntu120.04  |
| +1 :green_heart: |  javadoc  |   1m 41s |  |  the patch passed with JDK Private Build-1.8.0_342-8u342-b07-0ubuntu1~20.04-b07  |
| +1 :green_heart: |  spotbugs  |   4m 36s |  |  the patch passed  |
| +1 :green_heart: |  shadedclient  |  20m 20s |  |  patch has no errors when building and testing our client artifacts.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   7m 15s |  |  hadoop-mapreduce-client-core in the patch passed.  |
| +1 :green_heart: |  unit  |   8m 51s |  |  hadoop-mapreduce-client-app in the patch passed.  |
| +1 :green_heart: |  unit  |   5m  0s |  |  hadoop-mapreduce-client-hs in the patch passed.  |
| +1 :green_heart: |  unit  | 140m 33s |  |  hadoop-mapreduce-client-jobclient in the patch passed.  |
| +1 :green_heart: |  asflicense  |   1m  1s |  |  The patch does not generate ASF License warnings.  |
|  |   | 293m 16s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-4980/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/4980 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient spotbugs checkstyle codespell detsecrets |
| uname | Linux 639dd1f5bc01 4.15.0-191-generic #202-Ubuntu SMP Thu Aug 4 01:49:29 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | trunk / c2a34a2cd432555fa79e050f1b3eaf774a233bcd |
| Default Java | Private Build-1.8.0_342-8u342-b07-0ubuntu1~20.04-b07 |
| Multi-JDK versions | /usr/lib/jvm/java-11-openjdk-amd64:Ubuntu-11.0.16+8-post-Ubuntu-0ubuntu120.04 /usr/lib/jvm/java-8-openjdk-amd64:Private Build-1.8.0_342-8u342-b07-0ubuntu1~20.04-b07 |
|  Test Results | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-4980/1/testReport/ |
| Max. process+thread count | 1613 (vs. ulimit of 5500) |
| modules | C: hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-core hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-app hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-hs hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-jobclient U: hadoop-mapreduce-project/hadoop-mapreduce-client |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-4980/1/console |
| versions | git=2.25.1 maven=3.6.3 spotbugs=4.2.2 |
| Powered by | Apache Yetus 0.14.0 https://yetus.apache.org |


This message was automatically generated.

**Review by hotcodemacha — APPROVED** [review]
+1 LGTM

File: hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-core/src/main/java/org/apache/hadoop/mapred/QueueConfigurationParser.java [hunk 1]
```diff
@@ -88,7 +91,7 @@ class QueueConfigurationParser {
   static final String VALUE_TAG = "value";
 
   /**
-   * Default constructor for DeperacatedQueueConfigurationParser
+   * Default constructor for QueueConfigurationParser
```
Comment (steveloughran) [inline hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-core/src/main/java/org/apache/hadoop/mapred/QueueConfigurationParser.java]: can you add a . at the end if you are going near this line

File: hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-jobclient/src/test/java/org/apache/hadoop/mapred/TestQueueConfigurationParser.java [hunk 2]
```diff
@@ -28,12 +28,13 @@
 import javax.xml.transform.dom.DOMSource;
 import javax.xml.transform.stream.StreamResult;
 
-import org.w3c.dom.Document;
```
Comment (steveloughran) [inline hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-jobclient/src/test/java/org/apache/hadoop/mapred/TestQueueConfigurationParser.java:31]: just leave these imports alone; it makes backporting harder than it needs to be

**Review by steveloughran — CHANGES_REQUESTED** [review]
two minor comments. the javadoc one is what comes of making a change to the comment; the import one because I do want to backport these and have just been dealing with the pain of the hdfs equivalent

File: hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-core/src/main/java/org/apache/hadoop/mapred/QueueConfigurationParser.java (hunk 1 above)
Comment (pjfanning) [reply hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-core/src/main/java/org/apache/hadoop/mapred/QueueConfigurationParser.java]: updated

File: hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-jobclient/src/test/java/org/apache/hadoop/mapred/TestQueueConfigurationParser.java (hunk 2 above)
Comment (pjfanning) [reply hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-jobclient/src/test/java/org/apache/hadoop/mapred/TestQueueConfigurationParser.java:31]: reverted some of the import changes

**Review by steveloughran — APPROVED** [review]
+1 pending a successful yetus run -something is playing up there so try a rebase and force push to see if that helps

**Comment by steveloughran** [comment]
@pjfanning can you rebase and push up to kick jenkins off again? thanks

**Comment by pjfanning** [comment]
@steveloughran rebased and build restarted

**Comment by hadoop-yetus** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 51s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  1s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  1s |  |  detect-secrets was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  |  The patch appears to include 12 new or modified test files.  |
|||| _ trunk Compile Tests _ |
| +0 :ok: |  mvndep  |  15m 41s |  |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |  25m 43s |  |  trunk passed  |
| +1 :green_heart: |  compile  |   2m 50s |  |  trunk passed with JDK Ubuntu-11.0.16+8-post-Ubuntu-0ubuntu120.04  |
| +1 :green_heart: |  compile  |   2m 24s |  |  trunk passed with JDK Private Build-1.8.0_342-8u342-b07-0ubuntu1~20.04-b07  |
| +1 :green_heart: |  checkstyle  |   1m 16s |  |  trunk passed  |
| +1 :green_heart: |  mvnsite  |   3m 31s |  |  trunk passed  |
| +1 :green_heart: |  javadoc  |   2m 58s |  |  trunk passed with JDK Ubuntu-11.0.16+8-post-Ubuntu-0ubuntu120.04  |
| +1 :green_heart: |  javadoc  |   2m 41s |  |  trunk passed with JDK Private Build-1.8.0_342-8u342-b07-0ubuntu1~20.04-b07  |
| +1 :green_heart: |  spotbugs  |   4m 54s |  |  trunk passed  |
| +1 :green_heart: |  shadedclient  |  20m 40s |  |  branch has no errors when building and testing our client artifacts.  |
|||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 30s |  |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   2m 15s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   2m 34s |  |  the patch passed with JDK Ubuntu-11.0.16+8-post-Ubuntu-0ubuntu120.04  |
| +1 :green_heart: |  javac  |   2m 34s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   2m 11s |  |  the patch passed with JDK Private Build-1.8.0_342-8u342-b07-0ubuntu1~20.04-b07  |
| +1 :green_heart: |  javac  |   2m 11s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| +1 :green_heart: |  checkstyle  |   0m 57s |  |  hadoop-mapreduce-project/hadoop-mapreduce-client: The patch generated 0 new + 31 unchanged - 2 fixed = 31 total (was 33)  |
| +1 :green_heart: |  mvnsite  |   2m 33s |  |  the patch passed  |
| +1 :green_heart: |  javadoc  |   1m 56s |  |  the patch passed with JDK Ubuntu-11.0.16+8-post-Ubuntu-0ubuntu120.04  |
| +1 :green_heart: |  javadoc  |   1m 39s |  |  the patch passed with JDK Private Build-1.8.0_342-8u342-b07-0ubuntu1~20.04-b07  |
| +1 :green_heart: |  spotbugs  |   4m 32s |  |  the patch passed  |
| +1 :green_heart: |  shadedclient  |  20m 12s |  |  patch has no errors when building and testing our client artifacts.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   7m  9s |  |  hadoop-mapreduce-client-core in the patch passed.  |
| +1 :green_heart: |  unit  |   8m 47s |  |  hadoop-mapreduce-client-app in the patch passed.  |
| +1 :green_heart: |  unit  |   5m  7s |  |  hadoop-mapreduce-client-hs in the patch passed.  |
| +1 :green_heart: |  unit  | 134m 57s |  |  hadoop-mapreduce-client-jobclient in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 54s |  |  The patch does not generate ASF License warnings.  |
|  |   | 282m 40s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-4980/4/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/4980 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient spotbugs checkstyle codespell detsecrets |
| uname | Linux 2892e6611423 4.15.0-191-generic #202-Ubuntu SMP Thu Aug 4 01:49:29 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | trunk / 975f92cfdf1920c1dd709d3e3ce4471e94d86f40 |
| Default Java | Private Build-1.8.0_342-8u342-b07-0ubuntu1~20.04-b07 |
| Multi-JDK versions | /usr/lib/jvm/java-11-openjdk-amd64:Ubuntu-11.0.16+8-post-Ubuntu-0ubuntu120.04 /usr/lib/jvm/java-8-openjdk-amd64:Private Build-1.8.0_342-8u342-b07-0ubuntu1~20.04-b07 |
|  Test Results | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-4980/4/testReport/ |
| Max. process+thread count | 1590 (vs. ulimit of 5500) |
| modules | C: hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-core hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-app hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-hs hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-jobclient U: hadoop-mapreduce-project/hadoop-mapreduce-client |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-4980/4/console |
| versions | git=2.25.1 maven=3.6.3 spotbugs=4.2.2 |
| Powered by | Apache Yetus 0.14.0 https://yetus.apache.org |


This message was automatically generated.

**Pull request merged**